helpers/formValidation.tsx

```diff
@@ -0,0 +1,29 @@
+import * as Yup from 'yup'
+
+const TEXT_MIN = 2
+const TEXT_MAX = 20
+const PASSWORD_MIN = 6
+
+const signupValidation = Yup.object({
+  email: Yup.string()
+    .email('Invalid email address')
+    .required('Required'),
+  userName: Yup.string()
+    .min(TEXT_MIN, `Must be at least ${TEXT_MIN} characters`)
+    .max(TEXT_MAX, `Must be ${TEXT_MAX} characters or less`)
+    .required('Required'),
```

Comment:
To be consistent with the current naming scheme, I think we should enforce the first letter of the username be a lowercase letter.

There should be a check that only alphanumeric characters are used.

Reply:
I will update it with `string.lowercase()`

Reply:
one thing that really got me in c0d3.com codebase is.... we must decide early on.... `userName` as key or `username`. our old codebase has the cases different everywhere, became really painful.

Reply:
I would say we should use the camelCase convention ‘user name’ it’s 2 words then ‘userName’ make more sens in my opinion no ?

Reply:
I would say we should use the camelCase convention ‘user name’ it’s 2 words then ‘userName’ make more sens in my opinion no ?

Reply:
@hwong0305 PR updated feel free to merge it  if you are ok
About `username` I prefer to set everything in lower case instead of just the first char. _(we can always change it later if wanted)_

Reply:
@Wolfy64 officially its `username`, not `userName`. `userName` implies `user's name`, which is firstname + lastname usually.

proof: https://english.stackexchange.com/questions/43436/username-user-name-or-user-name

Reply:
I would not have believed it thank for the link